modules/exoplanets/exoplanetsmodule_lua.inl

```diff
@@ -56,120 +56,125 @@ constexpr const char* NoDataTextureFile =
 constexpr const char* DiscTextureFile =
     "${SYNC}/http/exoplanets_textures/1/disc_texture.png";
 
-void createExoplanetSystem(std::string_view starName) {
-    // If user have given name as in EOD, change it to speck-name
-    const std::string starNameSpeck = std::string(speckStarName(starName));
-
-    const std::string starIdentifier = createIdentifier(starNameSpeck);
-
-    std::string sanitizedStarName = starNameSpeck;
-    sanitizeNameString(sanitizedStarName);
-
-    const std::string guiPath = ExoplanetsGuiPath + sanitizedStarName;
-
-    SceneGraphNode* existingStarNode = sceneGraphNode(starIdentifier);
-    if (existingStarNode) {
-        LERROR(fmt::format(
-            "Adding of exoplanet system '{}' failed. The system has already been added.",
-            starName
-        ));
-        return;
-    }
+ExoplanetSystem findExoplanetSystemInData(std::string_view starName) {
+    ExoplanetSystem system;
 
     std::ifstream data(absPath(ExoplanetsDataPath), std::ios::in | std::ios::binary);
-
     if (!data.good()) {
         LERROR(fmt::format(
             "Failed to open exoplanets data file: '{}'", absPath(ExoplanetsDataPath)
         ));
-        return;
+        return ExoplanetSystem();
     }
 
     std::ifstream lut(absPath(LookUpTablePath));
     if (!lut.good()) {
         LERROR(fmt::format(
             "Failed to open exoplanets look-up table: '{}'", absPath(LookUpTablePath)
         ));
-        return;
+        return ExoplanetSystem();
     }
 
     // 1. search lut for the starname and return the corresponding location
     // 2. go to that location in the data file
     // 3. read sizeof(exoplanet) bytes into an exoplanet object.
-    Exoplanet p;
+    ExoplanetDataEntry p;
     std::string line;
-    bool found = false;
-
-    std::vector<Exoplanet> planetSystem;
-    std::vector<std::string> planetNames;
-
     while (getline(lut, line)) {
         std::istringstream ss(line);
         std::string name;
         getline(ss, name, ',');
 
-        if (name.compare(0, name.length() - 2, starNameSpeck) == 0) {
+        if (name.compare(0, name.length() - 2, starName) == 0) {
```

Comment:
I generally dislike the `compare` function since it's not super obvious which way the comparsion goes (does `== 0` mean that the strings are equal, unequal  since 0 normally means `false` but `==` means they they are equal), so i would suggest:
`if (name.substr(0, name.length() - 2) == starName) {`
instead